### ISSUE TITLE
Possible false positive on Nested #()s

### ISSUE THREAD

**Issue by hden** [clj-kondo/clj-kondo#1300]
**version**
`clj-kondo v2021.06.18`

**macro usage**
`clojure.test/are`

**platform**

```
; java --version
openjdk 11.0.10 2021-01-19
OpenJDK Runtime Environment GraalVM CE 20.3.1 (build 11.0.10+8-jvmci-20.3-b09)
OpenJDK 64-Bit Server VM GraalVM CE 20.3.1 (build 11.0.10+8-jvmci-20.3-b09, mixed mode, sharing)
```

**problem**

Possible false positive on #636

`Nested #()s are not allowed`

**repro**

`clj-kondo --lint test.clj`

```test.clj
(ns test
  (:require [clojure.test :refer [deftest testing are]]))

(def coll [])

(deftest linter-tests
  (testing "Nested #()s are not allowed"
    (are [key pred] (every? #(pred (get % key)) coll)
      :events #(every? (fn [{:keys [id timestamp]}]
                         (and (string? id)
                              (pos-int? timestamp)))
                       %))))

```

**config**

```
{:linters
 {:unresolved-symbol
  {:exclude [(clojure.test/are [thrown? thrown-with-msg? thrown-with-data?])
             (clojure.test/is [thrown? thrown-with-msg? thrown-with-data?])
             (io.pedestal.interceptor.error/error-dispatch [ctx ex])]}}
 :lint-as {superlifter.api/def-fetcher      clojure.core/defrecord
           superlifter.api/def-superfetcher clojure.core/defrecord
           diehard.core/defretrypolicy      clojure.core/def
           mojito.test-util/with-system     clojure.core/let}}
```

**Comment by borkdude** [comment]
Fixed on master.
